File: nasm/extendedEuclid.py
def ext_gcd(a,b):
    m, n = a, b
    xm, ym = 1, 0
    xn, yn = 0, 1
    while (n != 0):
        q = m // n # chia lấy phần nguyên
        r = m % n # chia lấy phần dư
        xr, yr = xm - q*xn, ym - q*yn
        m = n
        xm, ym = xn, yn
        n = r
        xn, yn = xr, yr
    return (m, xm, ym) # m = gcd(a,b) = xm * a + ym * b
a = int(input("hãy nhập số a: "))
b = int(input("hãy nhập số b: "))
print("=========================================================chúc Monstercuong7 một ngày tốt lành <3==============================")
print("(ước chung lớn nhất, x, y) lần lượt là: ",ext_gcd(a,b))

File: nasm/extendedEuclid.c
#include <stdio.h>

void extendedEuclid(int a, int b, int* gcd, int* x, int* y) {
    // Trường hợp cơ sở
    if (b == 0) {
        *gcd = a;
        *x = 1;
        *y = 0;
    }
    else {
        int x1, y1;
        extendedEuclid(b, a % b, gcd, &x1, &y1);
        *x = y1;
        *y = x1 - (a / b) * y1;
    }
}

int main() {
    int a, b, gcd, x, y;

    printf("Nhập số a: ");
    scanf("%d", &a);
    printf("Nhập số b: ");
    scanf("%d", &b);

    extendedEuclid(a, b, &gcd, &x, &y);

    printf("Ước số chung lớn nhất (GCD) của %d và %d là %d\n", a, b, gcd);
    printf("Các hệ số Bézout x và y là %d và %d\n", x, y);

    return 0;
}

